### DISCUSSION TITLE
How to import video file with geo location?

### DISCUSSION THREAD

**Discussion by kerkhc** [photoprism/photoprism#4132] [Q&A]
I imported several files, mostly pictures.
For imported videos (mov) I see a stack containing a new JPG for web UI. Somehow date/time has been set on the stack.
What kind of sidecar (json?, yaml?) would be suitable to index the video with appropriate date/time and geo-location?

**Comment by kerkhc** [comment]
found my answer by more searching ...

**Reply by aktentasche** [reply]
can you share what you found?

**Reply by kerkhc** [reply]
As a data base for meta data I prepared an calc sheet (Libre Office) containing GPS locations and time stamps related to either directories or files or file patterns. Some python scripts run through the sheet data and generate the YAML files (3rd approach - my preferred one). You could easily adjust the python scripts to generate JSON-files (2nd approach). Python examples can be found using some search engine. You may want to generate such script using ChatGPT.

There are different approaches. I researched on these three:

1. Generate jpeg-file:
Name the file as [movie-file].jpg, , e.g. _my-first-birthday.mov.jpg_
If jpeg-file is not Provided, photoprism will generate this one as a sidecar file.
Photoprism will take metadata from this jpeg-file.
I will not describe how to populate the jpeg-file with appropriate metadata. You may use exiftools or Python library to do so.

2. Generate json-file in Google TakeOut format:
Name the file as [movie-file].json , e.g. _my-first-birthday.mov.json_
Example:
```{
  "title": "this is my title - overwritten by photoprism",
  "description": "my description",
  "creationTime": {
    "timestamp": "1610432221",
    "formatted": "12.01.2021, 06:17:01 UTC"
  },
  "photoTakenTime": {
    "timestamp": "1384516978",
    "formatted": "15.11.2013, 12:02:58 UTC"
  },
  "geoData": {
    "latitude": 49.8416667,
    "longitude": 8.351111099999999,
    "altitude": 140.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "geoDataExif": {
    "latitude": 49.8416667,
    "longitude": 8.351111099999999,
    "altitude": 140.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  }
}
```
3. Generate photoprism sidecar file (YAML):
Name the file as [movie-file].yml, e.g. _my-first-birthday.mov.yml_
This is the approach that I finally applied. Example file:
```
Description: this is my description III.
DescriptionSrc: manual
TakenAt: 2016-09-01T00:00:00Z
TakenAtLocal: 2016-09-01T00:00:00Z
TakenSrc: manual
Lat: 45.36702975541012
Lng: 13.546802801497812
Altitude: 0.0
Year: 2016
Month: 09
Day: 01
```
